### PR TITLE
[mle] Add Status TLV check in Child Update Request and Response

### PR DESCRIPTION
**Pull request by jciupis**

### CODE REVIEW

File: src/core/thread/mle.cpp [hunk 1]
```diff
@@ -2859,6 +2860,18 @@ otError Mle::HandleChildUpdateRequest(const Message &aMessage, const Ip6::Messag
...
     SuccessOrExit(error = HandleLeaderData(aMessage, aMessageInfo));
 
+    // Status
+    if (Tlv::GetTlv(aMessage, Tlv::kStatus, sizeof(status), status) == OT_ERROR_NONE)
+    {
+        VerifyOrExit(status.IsValid(), error = OT_ERROR_PARSE);
+
+        if (status.GetStatus() == StatusTlv::kError)
+        {
+            BecomeDetached();
```
Comment (jwhui) [inline src/core/thread/mle.cpp]: We should validate that this message actually came from the parent by checking that the IPv6 source address matches the parent's link-local IPv6 address.

File: src/core/thread/mle_router.cpp [hunk 2]
```diff
@@ -2445,6 +2446,18 @@ otError MleRouter::HandleChildUpdateResponse(const Message &aMessage, const Ip6:
...
     }
 
+    // Status
+    if (Tlv::GetTlv(aMessage, Tlv::kStatus, sizeof(status), status) == OT_ERROR_NONE)
+    {
+        VerifyOrExit(status.IsValid(), error = OT_ERROR_PARSE);
+
+        if (status.GetStatus() == StatusTlv::kError)
+        {
+            RemoveStoredChild(child->GetRloc16());
```
Comment (jwhui) [inline src/core/thread/mle_router.cpp]: Should also set the child state to invalid.

File: src/core/thread/mle_router.cpp [hunk 3]
```diff
@@ -2445,6 +2446,18 @@ otError MleRouter::HandleChildUpdateResponse(const Message &aMessage, const Ip6:
                      error = OT_ERROR_SECURITY);
     }
 
+    // Status
+    if (Tlv::GetTlv(aMessage, Tlv::kStatus, sizeof(status), status) == OT_ERROR_NONE)
```
Comment (jwhui) [inline src/core/thread/mle_router.cpp]: I like how this is going.  However, OT does not currently implement a Child sending an MLE Child Update Response with Status TLV.  Are you also planning to add this feature?

Comment (jciupis) [reply src/core/thread/mle_router.cpp]: Such a message was sent by other vendor's stack during certification tests. I think that currently adding the sending feature is not necessary, but it's safer to process such a message if received.

**Pull request merged**